kafka-client/src/main/java/dev/responsive/kafka/internal/db/LwtWriterFactory.java

```diff
@@ -16,208 +16,154 @@
-  public static <K> LwtWriterFactory<K> reserve(
-      final RemoteTable<K, BoundStatement> table,
-      final CassandraClient client,
-      final int[] partitions,
-      final int kafkaPartition,
-      final long epoch,
-      final boolean windowed
-  ) {
-    for (final int sub : partitions) {
-      final var setEpoch = windowed
-          ? reserveEpochWindowed(table, client, sub, epoch)
-          : reserveEpoch(table,  client, sub, epoch);
-
-      if (!setEpoch.wasApplied()) {
-        final long otherEpoch = table.metadata(sub).epoch;
-        final var msg = String.format(
-            "Could not initialize commit buffer %s[%d] - attempted to claim epoch %d, "
-                + "but was fenced by a writer that claimed epoch %d on sub partition %d",
-            table.name(),
-            kafkaPartition,
-            epoch,
-            otherEpoch,
-            sub
-        );
-        final var e = new TaskMigratedException(msg);
-        LOG.warn(msg, e);
-        throw e;
-      }
-    }
-
-    return new LwtWriterFactory<>(
-        table,
-        epoch,
-        windowed
-            ? ensureEpochWindowed(table, client, epoch)
-            : ensureEpoch(table, client, epoch)
-    );
+  @Override
+  public String tableName() {
+    return table.name();
   }
 
-  public LwtWriterFactory(
-      final RemoteTable<K, BoundStatement> table,
-      final long epoch,
-      final PreparedStatement ensureEpoch
-  ) {
-    this.table = table;
-    this.epoch = epoch;
-    this.ensureEpoch = ensureEpoch;
+  @Override
+  protected P tablePartitionForKey(final K key) {
+    return partitioner.tablePartition(kafkaPartition, key);
   }
 
   @Override
-  public RemoteWriter<K> createWriter(
-      final SessionClients client,
-      final int partition
-  ) {
-    return new LwtWriter<>(
-        client.cassandraClient(),
-        () -> ensureEpoch.bind().setInt(PARTITION_KEY.bind(), partition),
-        table,
-        partition
-    );
+  public RemoteWriteResult<P> setOffset(final long offset) {
+    final P tablePartition = partitioner.metadataTablePartition(kafkaPartition);
+
+    final BatchStatementBuilder builder = new BatchStatementBuilder(BatchType.UNLOGGED);
+    builder.addStatement(fencingStatement(tablePartition));
+    builder.addStatement(table.setOffset(kafkaPartition, offset));
+
+    final var result = client.execute(builder.build());
+    return result.wasApplied()
+        ? RemoteWriteResult.success(tablePartition)
+        : RemoteWriteResult.failure(tablePartition);
   }
 
   @Override
-  public String toString() {
-    return "LwtWriterFactory{"
-        + "epoch=" + epoch
-        + '}';
+  protected long offset() {
+    return table.fetchOffset(kafkaPartition);
   }
 
-  private static ResultSet reserveEpoch(
-      final RemoteTable<?, BoundStatement> table,
-      final CassandraClient client,
-      final int partition,
-      final long epoch
+  @Override
+  public RemoteWriteResult<P> commitPendingFlush(
```

Comment:
ditto: why is the factory doing this?

Reply:
I think this is also a case of "it just needs to be renamed", but let me know if you had a more structural concern

(I filed a ticket to rename it so I won't forget)